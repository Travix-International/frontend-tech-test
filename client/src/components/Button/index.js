import "./styles.scss";
import React from "react";

const Button = ({ className, primary, children, ...rest }) => {
  const classNames = `button ${primary ? "button_primary" : ""} ${className}`;

  return (
    <button className={classNames} {...rest}>
      {children}
    </button>
  );
};

export default Button;
