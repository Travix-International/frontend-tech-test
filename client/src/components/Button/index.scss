@import '../../styles/theme.scss';

.btn {
    > i,
    > span {
        display: inline-block;
        pointer-events: none;
    }
    font-size: 1rem;
    line-height: 2rem;
    border-radius: 10px;
    padding: .2rem 1rem;
    font-weight: bold;
    transition: all .3s;
    outline: none;

    &:hover {
        cursor: pointer;
    }
}

.normal {
    color: $color-text;
    background-color: transparent;
    border: 1px solid $color-primary;
}

.primary {
    color: #fff;
    background-color: $color-primary;
    border: 1px solid $color-primary;

    &:hover {
        background-color: lighten($color-primary, 10%);
        border: 1px solid lighten($color-primary, 10%);
    }
}

.icon {
    border-radius: none;
    border: none;
    background-color: transparent;
    color: lighten($color-accent, 20%);
    &:hover{
        color: $color-accent;
    }
}