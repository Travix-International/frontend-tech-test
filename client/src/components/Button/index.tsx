import React from 'react';
import styles from './index.scss';

type ButtonType = 'primary' | 'normal' | 'icon';

interface Props {
    type?: ButtonType;
    title?: string;
    onClick?: () => void;
}

const Button: React.FC<Props> = ({ type = 'normal', title, onClick, children }) => {
    const handleClick = (e: React.MouseEvent<HTMLElement>) => {
        if (onClick) { onClick(); }
    };

    return (
        <button title={title} className={`${styles.btn} ${styles[type as ButtonType]}`} onClick={handleClick}>{children}</button>
    );
};

export default Button;