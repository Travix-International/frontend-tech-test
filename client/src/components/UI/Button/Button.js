import React from 'react'
import PropTypes from 'prop-types'

import styles from './Button.scss'

const Button = props => {
  return (
    <button
      className={`${props.add ? styles.Button_Add : null} ${
        props.delete ? styles.Button_Delete : null
      }`}
    >
      {props.text}
    </button>
  )
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  add: PropTypes.bool,
  delete: PropTypes.bool
}

export default Button
