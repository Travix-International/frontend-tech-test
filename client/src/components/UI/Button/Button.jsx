import React from 'react';

import styles from './Button.module.scss';

const Button = props => (
  <button {...props} className={styles.Button}>{props.children}</button>
);

export default Button;