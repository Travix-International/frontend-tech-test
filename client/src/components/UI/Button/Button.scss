@import '../../../scss/vars.scss';
.Button {
  position: relative;
  overflow: hidden;
  margin: 1rem;
  padding: 1rem;
  font-size: 1rem;
  text-transform: uppercase;
  border-radius: 1rem;
  border: none;
  color: white;
  transition: all 0.5s ease-in-out;
  outline: none;
  &:active:after {
    padding: 0;
    margin: 0;
    opacity: 1;
    transition: 0s
  }
  &:after {
    content: "";
    display: block;
    position: absolute;
    border-radius: 50%;
    padding-top: 240%;
    padding-left: 240%;
    margin-left: -120%;
    margin-top: -120%;
    opacity: 0;
    transition: all 0.8s
  }
  &_Add {
    @extend .Button;
    background-color: $darkgreen;
    &:hover {
      background-color: lighten($darkgreen, 15%);
      font-size: 2rem;
      transition: all 0.5s ease-in-out;
      cursor: pointer;
    }
    &:after {
      background-color: darken($darkgreen, 15%)
    }
  }
  &_Delete {
    @extend .Button;
    background-color: $red;
    &:hover {
      background-color: lighten($red, 15%);
      font-size: 2rem;
      transition: all 0.5s ease-in-out;
      cursor: pointer;
    }
    &:after {
      background-color: darken($red, 15%)
    }
  }
}