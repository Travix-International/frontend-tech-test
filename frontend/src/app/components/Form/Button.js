import React from 'react';
import PropTypes from 'prop-types';
import { FaCircleNotch } from 'react-icons/fa';
import { InputWrapper, Button as StyledButton } from './assets/style';

const Button = ({ type, id, isSubmiting, children, onClick, disabled }) => (
  <InputWrapper>
    <StyledButton type={type} id={id} onClick={() => onClick()} disabled={isSubmiting || disabled}>
      {isSubmiting ? <FaCircleNotch className="icon" /> : null} {children}
    </StyledButton>
  </InputWrapper>
);

Button.defaultProps = {
  isSubmiting: false,
  disabled: false
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  isSubmiting: PropTypes.bool,
  disabled: PropTypes.bool
};

export default Button;
