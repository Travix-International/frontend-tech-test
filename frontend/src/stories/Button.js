import React from 'react';
import { storiesOf } from '@storybook/react';
import { Button } from '../app/components/Form';

storiesOf('Button', module)
  .add('primary', () => (
    <Button id="some-id" type="primary">
      Primary
    </Button>
  ))
  .add('Secondary', () => (
    <Button id="some-id" type="secondary">
      Secondary
    </Button>
  ))
  .add('Danger', () => (
    <Button id="some-id" type="danger">
      Danger
    </Button>
  ))
  .add('Disabled', () => (
    <Button id="some-id" type="primary" disabled>
      Disabled
    </Button>
  ))
  .add('Submitting', () => (
    <Button id="some-id" type="primary" isSubmiting>
      Submitting
    </Button>
  ));
