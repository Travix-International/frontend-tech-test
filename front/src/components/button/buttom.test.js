import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Button from '.'

describe('ButtonComponent', () => {
  it('Snapshot', async () => {
    const tree = renderer
      .create(<Button>Test</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('onCLick simulate', async () => {
    const onClick = jest.fn()
    const tree = shallow(<Button onClick={onClick}>Test</Button>)
    tree.find('Button').simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
