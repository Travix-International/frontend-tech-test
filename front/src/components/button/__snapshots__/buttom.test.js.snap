// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent Snapshot 1`] = `
.c0 {
  border: 0;
  padding: 0;
  color: #fff;
  background-color: #2196f3;
  margin: .8rem;
  box-shadow: 0 .1rem .5rem 0 rgba(0,0,0,0.2),0 .2rem .2rem 0 rgba(0,0,0,0.14),0 .3rem .1rem -0.2rem rgba(0,0,0,0.12);
  padding: .6rem 1.6rem;
  font-size: 1rem;
  min-width: 6.4rem;
  box-sizing: border-box;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  line-height: 1.75;
  border-radius: .4rem;
  text-transform: uppercase;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  background-color: #1976d2;
}

.c0:active {
  box-shadow: 0 .5rem .5rem -0.3rem rgba(0,0,0,0.2),0 .8rem 1rem .1rem rgba(0,0,0,0.14),0 .3rem 1.4rem .2rem rgba(0,0,0,0.12);
}

<button
  className="c0"
>
  Test
</button>
`;
